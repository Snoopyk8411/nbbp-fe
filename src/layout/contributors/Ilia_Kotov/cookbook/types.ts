import { NextPage } from 'next';

export type NextPageWithLayout<Props> = NextPage<Props> & {
  getLayout?: (page: NextPage) => JSX.Element;
};
