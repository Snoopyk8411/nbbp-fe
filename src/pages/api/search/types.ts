import { IProduct } from 'tools/types/api-product-types';

export type ISearchKeysType = 'title';

export interface ISearchResult {
  item: IProduct;
}
