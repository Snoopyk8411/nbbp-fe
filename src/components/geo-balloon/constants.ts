export const BUTTON_TEXT = 'Выбрать';
