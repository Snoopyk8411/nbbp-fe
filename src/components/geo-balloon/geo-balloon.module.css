.wrapper {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
}

.heading {
  color: var(--clr-neutral-700);
  font-size: 1.25rem;
  font-weight: 700;
}

.description {
  color: var(--clr-neutral-700);
}

.button {
  display: flex;
  justify-content: space-around;
  align-items: center;
  border-radius: 0.25rem;
  padding: 0.5rem 1rem;
  user-select: none;
  cursor: pointer;
  outline: none;
  border: none;
  color: var(--clr-neutral-700);
  background-color: var(--clr-neutral-100);
  align-self: flex-end;
}
