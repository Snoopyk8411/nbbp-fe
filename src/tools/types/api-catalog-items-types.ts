// Catalog item (link) appearance
export enum Appearance {
  PLAIN, // default styles
  ACCENT, // color accent applied
  HIGHLIGHT, // highlighted
}

export interface ICatalogItem {
  id: string;
  name: string;
  url: string;
  groups: string[];
  appearance: Appearance;
  svg?: string;
}

export type ICatalogNode = Omit<ICatalogItem, 'groups'> & { order: number };

export interface ICatalogTree {
  node: ICatalogNode;
  children: ICatalogTree[];
}

export enum Errors {
  WRONG_TYPE = 'Category type should be string',
  NOT_FOUND = 'Category not found',
}
