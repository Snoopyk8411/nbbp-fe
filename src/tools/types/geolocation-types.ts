export type ICoords = [number, number];

export interface IGeoPoint {
  id: string;
  title: string;
  tag: string;
  description: string;
  coords: ICoords;
}
