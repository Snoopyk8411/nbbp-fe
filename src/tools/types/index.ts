/**
 * Usage:
 * interface PageData {
 *  pageType: PageType;
 *  pageId?: string;
 *  id?: string;
 * }
 * and we need or 'pageId' or 'id', and the second one is must be 'undefined' then we use it like so:
 *
 * type AnyPage = RequireOnlyOne<PageData, "id" | "pageId">;
 */

export type RequireOnlyOne<T, Keys extends keyof T = keyof T> = Pick<T, Exclude<keyof T, Keys>> &
  {
    [K in Keys]-?: Required<Pick<T, K>> & Partial<Record<Exclude<Keys, K>, undefined>>;
  }[Keys];
