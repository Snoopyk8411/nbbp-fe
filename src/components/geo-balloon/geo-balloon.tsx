import { FC } from 'react';

import { IGeoPoint } from 'tools/types/geolocation-types';
import balloonStyles from './geo-balloon.module.css';

interface IGeoBaloonProps {
  point: IGeoPoint;
  handleSelect?: () => void;
}

export const GeoBalloon: FC<IGeoBaloonProps> = ({ point, handleSelect }) => {
  const { title, description } = point;
  return (
    <div className={balloonStyles.wrapper}>
      <h4 className={balloonStyles.heading}>{title}</h4>
      <div>{description}</div>
      <button className={balloonStyles.button} onClick={handleSelect} type='button'>
        Выбрать
      </button>
    </div>
  );
};
