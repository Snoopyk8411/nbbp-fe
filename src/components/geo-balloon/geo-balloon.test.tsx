import renderer from 'react-test-renderer';

import { IGeoPoint } from 'tools/types/geolocation-types';
import { GeoBalloon } from './geo-balloon';

describe('Geo balloon component', () => {
  it('should get valid snapshot', () => {
    const point: IGeoPoint = {
      id: '1',
      title: 'Title',
      tag: 'Tag',
      description: 'Description',
      coords: [0, 0],
    };
    const component = renderer.create(<GeoBalloon point={point} />);
    expect(component).toMatchSnapshot();
  });
});
