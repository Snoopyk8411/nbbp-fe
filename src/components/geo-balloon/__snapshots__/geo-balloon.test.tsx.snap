// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Geo balloon component should get valid snapshot 1`] = `
<div
  className="wrapper"
>
  <h4
    className="heading"
  >
    Title
  </h4>
  <div>
    Description
  </div>
  <button
    className="button"
    type="button"
  >
    Выбрать
  </button>
</div>
`;
